packages/v4/src/content/design-guidelines/components/code-editor/code-editor.md

```diff
@@ -0,0 +1,56 @@
+---
+id: Code editor
+section: components
+---
+
+A **code editor** is a versatile text editor that allows for editing various languages. The code editor is built on top of the [Monaco Editor](https://microsoft.github.io/monaco-editor/).
+
+## Elements
+By default, the bulk of the code editor is the text editor itself. The component is extremely flexible, so many of the features are optional. Much of the functionality of the Monaco Editor has been carried over to the code editor, so it’s possible to further customize the component based on their options.
+
+### Light theme editor
+
+<img src="./img/basic-code-editor.png" alt="Example of a basic code editor and its elements" width="800"/>
+
+1. **Text editor:** The text of the editor sits in this container. Syntax highlighting is automatically built into the editor, but can be customized if necessary.
+2. **Language indicator (optional):** This tab indicates the language that the code is written in. 
+3. **Minimap (optional):** This displays an overview of the contents of the editor. It can be used to scroll through the editor, rather than using the standard scrollbar that appears on the right of the editor.
+4. **Line numbers (optional):** These will populate to accompany the lines of text that appear in the editor.
+5. **Indentation indicators:** These vertical lines help make indentation clearer. They appear per indentation level.
+
+### Dark theme editor
+We recommend using the light theme editor by default, but there is also a dark theme code editor available. All editor functionality remains the same whether a light or dark theme is used.
+
+<img src="./img/code-editor-dark.png" alt="Example of a dark themed code editor" width="800"/>
+
+Syntax highlighting will update automatically to adapt to the dark background. The editor still remains within the code editor container so only the inner area adapts the dark theme.
+
+## Usage
+Use a code editor when you:
+1. Want the user to create or edit large blocks of code.
+2. Want to display large blocks of code (see [Displaying code in a code editor](https://docs.google.com/document/d/1mK5C_fflcFHbZak2D9QaUvUc6oLgpf8VwBcZulnVOjI/edit#heading=h.vd380d1q7bjm) for guidance on when to use a code editor over a [code block](/components/code-block)).
```

Comment:
Displaying code in a code editor should be [Displaying code in a code editor](#displaying-code-in-a-code-editor)